hibernate-core/src/main/java/org/hibernate/loader/plan/exec/internal/LoadQueryJoinAndFetchProcessor.java

```diff
@@ -222,16 +222,43 @@ private AssociationType getJoinedAssociationTypeOrNull(Join join) {
 				: null;
 	}
 
-	private String resolveAdditionalJoinCondition(String rhsTableAlias, String withClause, Joinable joinable, AssociationType associationType) {
+	private String resolveAdditionalJoinCondition(
+			Join join,
+			String rhsTableAlias,
+			String withClause,
+			Joinable joinable,
+			AssociationType associationType) {
 		// turns out that the call to AssociationType#getOnCondition in the initial code really just translates to
 		// calls to the Joinable.filterFragment() method where the Joinable is either the entity or
 		// collection persister
-		final String filter = associationType!=null?
-				associationType.getOnCondition( rhsTableAlias, factory, queryInfluencers.getEnabledFilters() ):
-				joinable.filterFragment(
+
+		final String filter;
+		// If the left hand side
+		if ( associationType.isEntityType() &&
+				join.getLeftHandSide().getDisposition() != QuerySpace.Disposition.COLLECTION ) {
+			// We always need to get the filter (if any) for the following cases:
+			// * composite with entity fetch
+			// * entity with entity fetch
+			// EntityType#getOnCondition doesn't always do the right thing in this case,
```

Comment:
So shouldn't we fix it instead? Or is it used in other situations?

Reply:
Changing EntityType#getOnCondition to work for this use case breaks other tests. The problem is that we we multiple ways of walking code in 5.x, and they seem to use EntityType#getOnCondition for different reasons.

@dreab8, is the walking code unified or Hibernate 6?

If so, we shouldn't spend a lot of resources on fixing this in Hibernate 5.

Reply:
OK, that is unfortunate but you're right, it's probably better to be cautious about this one.